packages/eventual-send/src/index.d.ts

```diff
@@ -95,79 +171,81 @@ declare interface HandledPromiseConstructor
   prototype: Promise<unknown>;
 }
 
-declare var HandledPromise: HandledPromiseConstructor;
-
 declare namespace global {
+  // eslint-disable-next-line vars-on-top,no-var
   var HandledPromise: HandledPromiseConstructor;
 }
 
-declare function makeHandledPromise(): HandledPromiseConstructor;
+export declare const HandledPromise: HandledPromiseConstructor;
+
+/**
+ * "E" short for "Eventual", what we call something that has to return a promise.
+ */
+type ECallable<T extends Callable> = ReturnType<T> extends PromiseLike<infer U>
+  ? T // function already returns a promise
+  : (...args: Parameters<T>) => Promise<Awaited<ReturnType<T>>>; // make it return a promise
 
 /* Types for E proxy calls. */
-type ESingleMethod<T> = {
-  readonly [P in keyof T]: (
-    ...args: Parameters<T[P]>
-  ) => Promise<Awaited<ReturnType<T[P]>>>;
+
+/**
+ * Transform each function in T to return a promise
+ */
+type EMethods<T> = {
+  readonly [P in keyof T]: T[P] extends Callable ? ECallable<T[P]> : never;
 };
-type ESingleCall<T> = T extends Function
-  ? ((...args: Parameters<T>) => Promise<Awaited<ReturnType<T>>>) &
-      ESingleMethod<Required<T>>
-  : ESingleMethod<Required<T>>;
-type ESingleGet<T> = {
-  readonly [P in keyof T]: Promise<Awaited<T[P]>>;
+
+type ECallableOrMethods<T> = T extends Callable
+  ? ECallable<T> & EMethods<Required<T>>
+  : EMethods<Required<T>>;
+
+type EGetters<T> = {
+  readonly [P in keyof T]: T[P] extends PromiseLike<infer U>
+    ? T[P]
+    : Promise<Awaited<T[P]>>;
 };
 
 /* Same types for send-only. */
-type ESingleMethodOnly<T> = {
-  readonly [P in keyof T]: (...args: Parameters<T[P]>) => void;
-};
-type ESingleCallOnly<T> = T extends Function
-  ? ((...args: Parameters<T>) => void) & ESingleMethodOnly<T>
-  : ESingleMethodOnly<T>;
-type ESingleGetOnly<T> = {
-  readonly [P in keyof T]: void;
+type ESendOnlyCallable<T extends Callable> = (
+  ...args: Parameters<T>
+) => Promise<void>;
+
+type ESendOnlyMethods<T> = {
+  readonly [P in keyof T]: T[P] extends Callable
+    ? ESendOnlyCallable<T[P]>
+    : never;
 };
 
+type ESendOnlyCallableOrMethods<T> = T extends Callable
+  ? ESendOnlyCallable<T> & ESendOnlyMethods<Required<T>>
+  : ESendOnlyMethods<Required<T>>;
+
 interface ESendOnly {
-  <T>(x: T): ESingleCallOnly<Awaited<T>>;
+  <T>(x: T): ESendOnlyCallableOrMethods<RemoteFunctions<T>>;
```

Comment:
this could also have a function overload.

if you don't want to include in this PR I can follow up

Reply:
> if you don't want to include in this PR I can follow up

Yes, please.  Let's chat about it before you spend much time; I think the proposals you made don't deal with `RemoteFunctions<T>` correctly.